lib/hydra/works/models/concerns/collection_behavior.rb

```diff
@@ -22,6 +22,39 @@ module CollectionBehavior
       type [RDFVocabularies::PCDMTerms.Collection,WorksVocabularies::WorksTerms.Collection]
       include Hydra::Works::AggregatesGenericWorks
       include Hydra::Works::AggregatesCollections
+      include Hydra::Works::BlockChildObjects
+
+      filters_association :members, as: :child_collections, condition: :works_collection?
+      filters_association :members, as: :child_generic_works, condition: :works_generic_work?
+    end
+
+    module ClassMethods
+      def indexer
+        Hydra::Works::CollectionIndexer
+      end
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Collection.
+    def works_collection?
```

Comment:
The way the code is written, there is a difference between a pcdm:collection and a works:collection.  A pcdm:collection aggregates pcdm:objects.  A works:collection aggregates works:collections and works:generic_works.

On an intuitive level, I am not convinced that this is exactly what we want, but haven't had time to work through examples of collections that mix use of pcdm:objects and works:generic_works.